### ISSUE TITLE
Deprecate Theme Switch modal because: "Preserve my homepage content" modal does not load preview for certain themes 

### ISSUE THREAD

**Issue by fushar** [Automattic/wp-calypso#69687]
# Description
[Updated 20 Jan 2023]
Over the past months we have had many discussions on how the visuals in  theme switch modal is "mostly" broken. It some very specific cases the live preview (the left side visual) works properly and applies the styles of the new theme to the user's existing content.  For the cases where this works properly we want to show the visuals.  Otherwise we want to hide the visuals and just show the copy in the modal.

See discussion in these P2s:
*  pekYwv-uV-p2
* [Auto loading homepage on theme switch issues] pbAok1-1Yh-p2

### Steps to reproduce

1. Create a new site (wordpress.com/start) with a theme whose homepage template is `home.html`
   - e.g.: Pendant / Cultivate / Archeo / Yuga
2. On My Home, open Appearances -> Themes.
3. Select a theme whose homepage template is not `home.html`
   - e.g.: Appleton / Dorna / Russell
4. Click "Activate this design"

### Expected

The previews for both column load

### What happened instead

The preview for the "Switch theme, preserving my homepage content." column does not load:

<img width="445" alt="image" src="https://user-images.githubusercontent.com/1525580/199427872-74b81fb4-f5ff-4484-bf7a-84f6b5ab2dc3.png">

# Tasks
- [ ] ~Show the modal for FSE -> FSE theme switch where XXXXX criteria to be decided~
- [ ] ~Hide the modal when the live preview visual is broken or shows something useless~
- [ ]  Hide the modal for all cases. See discussion  p1675816912909179-slack-C048CUFRGFQ.
- [ ] Inform  @ianstewart , @tanjoymor and HE before this ships to users p1674190501522309/1674172748.806669-slack-C048CUFRGFQ
- [ ] Write a short post on wpcom happy

@Automattic/lego

**Comment by fushar** [comment]
Update on investigation: I've narrowed down the possible culprits; the preview loads if I commented out the following two lines 😅 

- fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sjc%2Qgurzr%2Qcerivrj.cuc%3Se%3Q09r44316%23124-og
- fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sjc%2Qgurzr%2Qcerivrj.cuc%3Se%3Q09r44316%23128-og

Will investigate further...

**Comment by arthur791004** [comment]
It's a bug in the theme preview. See https://core.trac.wordpress.org/ticket/56908

**Comment by fushar** [comment]
Thanks @arthur791004!

Confirmed that by applying the patch that is proposed in the above issue, this broken theme preview issue is also resolved 😄 

<img width="744" alt="image" src="https://user-images.githubusercontent.com/1525580/199645853-838b507d-3b83-431d-86cd-245655d1ae43.png">

**Comment by autumnfjeld** [comment]
@ianstewart  Just an FYI on something fixed related to theme switch, to add to the theme switch audit knowledge. Work started weeks ago.

**Comment by autumnfjeld** [comment]
@Automattic/lego  Let's make sure this gets done in Sprint  29 (Feb 2023).

**Comment by miksansegundo** [comment]
@Automattic/lego Here is the [PR](https://github.com/Automattic/wp-calypso/pull/73726) to remove the modal.

I believe there are more reasons to deprecate this modal, not only because the preview for "Preserve my homepage content" does not load for certain themes.

I investigated the reasons why the modal doesn't work as expected: 

- The preview on the left side of the modal keeps loading because of a core bug. See the [ticket and the patch](https://core.trac.wordpress.org/ticket/56908) that still needs to be merged.
- The preview on the right side of the modal, which says "Replace my homepage content...", deceives users because it shows exactly the theme demo site. After the theme is activated, users realize their homepage doesn't look exactly like the preview they saw in the modal.
- The theme switch fails often due to missing or wrong data in the headstart annotations or [theme tags](https://github.com/Automattic/themes/blob/trunk/geologist/style.css#L15):  
   - when the theme's homepage wants to show a list of posts
     - the homepage template remains the same because themes that set `show_on_front: posts` in their headstart annotations also need to set:
       - the tag `blog-homepage` and `auto-loading-homepage` in the comments in the style.css file.
   - when the theme's homepage wants to show a `page` (with or without a `page_template`), the homepage remains the same because themes that set `show_on_front: page` in their headstart annotations also need to set: 
       - a `post-type: page` with `hs_custom_meta: _hs_front_page` in the `content` property of the annotation. 
       - the tag `auto-loading-homepage` in the comments in the style.css file.

I found that the logic that handles these homepage updates is in the class-template-first-themes.php. See fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sshyy%2Qfvgr%2Qrqvgvat%2Spynff%2Qgrzcyngr%2Qsvefg%2Qgurzrf.cuc-og

I'll record a video showing cases when the homepage switching fails because theme annotations or tags (theme features) are missing. 

**Related**
https://github.com/Automattic/themes/issues/6660
https://github.com/Automattic/themes/pull/3275 
https://github.com/Automattic/themes/pull/4874 
https://github.com/Automattic/wp-calypso/issues/64705 
https://github.com/Automattic/wp-calypso/issues/57133 

Here are some issues related to the misleading theme demo site preview: 
https://github.com/Automattic/themes/issues/5137 
https://github.com/Automattic/wp-calypso/issues/57670

**Comment by autumnfjeld** [comment]
I'll move this to the backlog for now.

**Comment by okmttdhr** [comment]
I confirmed it worked as expected when switching Pendant to Appleton on Apr 24, 2023.
<img width="752" alt="Screen Shot 2023-04-24 at 14 32 02" src="https://user-images.githubusercontent.com/5287479/233907829-4ad3a4a4-ac72-4755-9d8c-c65217b36edd.png">

>It's a bug in the theme preview. See [core.trac.wordpress.org/ticket/56908](https://core.trac.wordpress.org/ticket/56908)

I wonder if this ticket has been fixed, or am I missing other cases where this bug happens? 💭

**Comment by autumnfjeld** [comment]
@okmttdhr  I also followed the instructions in issue:

* Start with a theme whose homepage template is home.html
   * I started with a site that used blank canvas with a home.html 
* On My Home, open Appearances -> Themes.
* Select a theme whose homepage template is not home.html
* I chose Appleton
* Click "Activate this design"

<img width="742" alt="image" src="https://user-images.githubusercontent.com/5665959/233918054-dc7577c4-27bd-4c1c-84fe-811937d228f5.png">

We can close this issue.  We've referenced it in the spike P2 and issue, so it can be easily referred to.     

It would probably be good to create a grid somewhere (excel or in the P2) where we start to document in which cases the live preview works and which cases it does not.  Something like [this](https://docs.google.com/spreadsheets/d/1KNxYhsA_OlaT0wAZg1FzM2myVlaOH8C78aI43XxTsqI/edit#gid=0), but better.  A grid that shows all the permutations.  Maybe in a better tool?
